EventListener/Doctrine/FlushListener.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Vich\UploaderBundle\EventListener\Doctrine;
+
+use Doctrine\ORM\Event\PostFlushEventArgs;
+
+/**
+ * FlushListener.
+ *
+ * Listen to the flush event.
+ *
+ * @author Kim Wüstkamp <kim@wuestkamp.com>
+ */
+class FlushListener extends BaseListener
+{
+    /**
+     * The events the listener is subscribed to.
+     *
+     * @return array The array of events
+     */
+    public function getSubscribedEvents(): array
+    {
+        return [
+            'postFlush',
+        ];
+    }
+
+    /**
+     * Initiates deletion of to-be-removed files in queue.
+     *
+     * @param PostFlushEventArgs $event The event
+     */
+    public function postFlush(PostFlushEventArgs $event): void
+    {
+        $em = $event->getEntityManager();
+        $entities = $this->handler->removeFilesInQueue();
+
+        if (count($entities) > 0) {
+            foreach ($entities as $entity) {
```

Comment:
Are we really needing to persist here? Entities should be already in the entity manager

Reply:
You're correct, no persist needed.